feat: Добавить контроллеры, объектную модель, тесты, исключения

Hivludi opened the pull request.

Hivludi committed:
feat: Добавить контроллеры, объектную модель, тесты, исключения

jeggga commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,20 @@
+import javax.validation.constraints.NotNull;
+import java.time.LocalDate;
+
+@Data
```
Для полной перегрузки конструктора можно еще добавить следующие аннотации `@NoArgsConstructor
@AllArgsConstructor`

Hivludi replied:
AllArgsConstructor добавил, NoArgsConstructor потребует инициализации полей, так то пока его не стал добавлять

jeggga commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,21 @@
+import java.time.LocalDate;
+
+@Data
+@Builder(toBuilder = true)
```
Аналогичное замечание, как и у класса `Film`

jeggga commented on src/main/java/ru/yandex/practicum/filmorate/validator/FilmValidator.java:
```diff
@@ -0,0 +1,18 @@
+
+import java.time.LocalDate;
+
+public class FilmValidator {
```
Отлично, что решил валидацию выделить в отдельный класс (сервис)

jeggga commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,54 @@
+import java.util.Map;
+
+@RestController
+@RequestMapping("/films")
```
Отлично, что выделил корневую директорию

Hivludi committed:
feat: Добавить конструкторы со всеми полями для фильмов и юзеров

Hivludi committed:
refactor: Заменить классы валидации на аннотации спринга + кастомные для дат
feat: Добавить вывод причин ошибок валидации в логи и http-ответ
test: Заменить тесты старых валидаторов на тесты новых
